api/spec/src/entitlements/entitlements.tsp

```diff
@@ -0,0 +1,319 @@
+   */
+  @get
+  @operationId("listEntitlements")
+  @summary("List all entitlements")
+  list(
+    /**
+     * Filtering by multiple features.
+     *
+     * Usage: `?feature=feature-1&feature=feature-2`
+     */
+    @query feature?: string[],
+
+    /**
+     * Filtering by multiple subjects.
+     *
+     * Usage: `?subject=customer-1&subject=customer-2`
+     */
+    @query subject?: string[],
+
+    /**
+     * Filtering by multiple entitlement types.
+     *
+     * Usage: `?entitlementType=metered&entitlementType=boolean`
+     */
+    @query entitlementType?: EntitlementType[],
+
+    ...OpenMeter.QueryPagination,
+    ...OpenMeter.QueryLimitOffset,
+    ...OpenMeter.QueryOrdering<EntitlementOrderBy>,
+  ): Entitlement[] | OpenMeter.PaginatedResponse<Entitlement> | OpenMeter.CommonErrors;
+
+  /**
+   * Get entitlement by id.
+   */
+  @get
+  @operationId("getEntitlementById")
+  get(
+    @path
+    entitlementId: ULID,
+  ): Entitlement | OpenMeter.CommonErrors | OpenMeter.NotFoundError;
+}
+
+/**
+ * Type of the entitlement.
+ */
+@friendlyName("EntitlementType")
+enum EntitlementType {
+  metered: "metered",
+  boolean: "boolean",
+  static: "static",
+}
+
+// NOTE: use OmitProperties<T> for templates once entitlement models are defined
+
+/**
+ * Entitlement templates are used to define the entitlements of a plan.
+ * Features are omitted from the entitlement template, as they are defined in the rate card.
+ */
+@discriminator("type")
+@friendlyName("Entitlement")
+union Entitlement {
+  metered: EntitlementMetered,
+  static: EntitlementStatic,
+  boolean: EntitlementBoolean,
+}
+
+/**
+ * Shared fields for entitlement creation
+ */
+@friendlyName("EntitlementCreateSharedFields")
+model EntitlementCreateSharedFields {
+  /**
+   * The feature the subject is entitled to use.
+   * Either featureKey or featureId is required.
+   */
+  @example("example-feature-key")
+  featureKey?: Key;
+
+  /**
+   * The feature the subject is entitled to use.
+   * Either featureKey or featureId is required.
+   */
+  @example("01ARZ3NDEKTSV4RRFFQ69G5FAV")
+  featureId?: ULID;
+
+  /**
+   * Additional metadata for the feature.
+   */
+  metadata?: Metadata;
+
+  usagePeriod?: RecurringPeriod;
+}
+
+/**
+ * Create inpurs for metered entitlement
+ */
+@friendlyName("EntitlementMeteredCreateInputs")
+model EntitlementMeteredCreateInputs {
+  ...OmitProperties<EntitlementCreateSharedFields, "usagePeriod">;
+  type: EntitlementType.metered;
+
+  /**
+   * If softLimit=true the subject can use the feature even if the entitlement is exhausted, hasAccess will always be true.
+   */
+  isSoftLimit?: boolean = false;
+
+  /**
+   * Deprecated, ignored by the backend. Please use isSoftLimit instead; this field will be removed in the future.
+   */
+  @deprecated("Use isSoftLimit instead")
+  isUnlimited?: boolean = false;
+
+  usagePeriod: RecurringPeriod;
+  measureUsageFrom?: MeasureUsageFrom;
```

Comment:
Missing documentation.

Reply:
The named type has the doc, same as in the current version. I can add docs for these too if you want.

Reply:
Alright.